pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -186,26 +189,48 @@ public String listMetadataForOneSegment(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
       @ApiParam (value = "Name of segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
       @ApiParam(value = "realtime|offline", required = false) @QueryParam("type") String tableTypeStr
-  ) {
+  ) throws JSONException {
     segmentName = checkGetEncodedParam(segmentName);
     CommonConstants.Helix.TableType tableType = Constants.validateTableType(tableTypeStr);
+    return listSegmentMetadataInternal(tableName, segmentName, tableType);
+  }
+
+  private String listSegmentMetadataInternal(@Nonnull  String tableName, @Nonnull String segmentName,
```

Comment:
I don't think this helper method is necessary. We just need a check as following:
if ((tableType == null || tableType == OFFLINE) && _pinotHelixResourceManager.hasOfflineTable(tableName)) {
      String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(tableName);
      result.put(getSegmentMetaData(offlineTableName, segmentName, TableType.OFFLINE));
}
This can ensure the behavior is the same as before